packages/grafana-data/src/types/variables.ts

```diff
@@ -0,0 +1,68 @@
+import { ComponentType } from 'react';
+import { Observable } from 'rxjs';
+
+import {
+  DataQuery,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceApi,
+  DataSourceJsonData,
+  DataSourceOptionsType,
+  DataSourceQueryType,
+  QueryEditorProps,
+} from './datasource';
+
+export type VariableSupportType = 'standard' | 'custom' | 'datasource';
```

Comment:
I think the type becomes easier to use with enum. This may be a personal preference, but I find enums more discoverable at least with Webstorm intellisense.

Type:
![image](https://user-images.githubusercontent.com/2376619/97867006-5010f100-1d0d-11eb-9131-d78a63f5aa23.png)

Enum:
![image](https://user-images.githubusercontent.com/2376619/97866990-47201f80-1d0d-11eb-88ad-5cafaf793135.png)

Reply:
Switch to enums and changed interfaces to classes instead, so plugin authors automatically get correct type when selecting class to inherit. Let me know what you think.